source/slang/slang-emit-wgsl.cpp

```diff
@@ -1506,4 +1506,59 @@ void WGSLSourceEmitter::emitIntrinsicCallExprImpl(
         inOuterPrec);
 }
 
+void WGSLSourceEmitter::_maybeEmitInterpolationAttribute(IRInterpolationMode mode)
+{
+    bool interpolationTypeEmitted = true;
+    switch (mode)
+    {
+    case IRInterpolationMode::NoInterpolation:
+        m_writer->emit("@interpolate(flat");
+        break;
+    case IRInterpolationMode::NoPerspective:
+    case IRInterpolationMode::Linear:
+        m_writer->emit("@interpolate(linear");
+        break;
+    default:
+        interpolationTypeEmitted = false;
+        break;
+    }
+
+    if (interpolationTypeEmitted)
+    {
+        switch (mode)
+        {
+        case IRInterpolationMode::Sample:
+            m_writer->emit(", sample");
+            break;
+        case IRInterpolationMode::Centroid:
+            m_writer->emit(", centroid");
+            break;
+        default:
```

Comment:
I think this `switch` will always hit the `default` case here, because in order for `interpolationTypeEmitted` to be true, `mode` has to be be one of three cases that is non of the cases listed in this `switch`.
That seems like a coding mistake.